xarray/core/variable.py

```diff
@@ -1782,7 +1784,13 @@ def reduce(
 
         if keep_attrs is None:
             keep_attrs = _get_keep_attrs(default=False)
-        attrs = self._attrs if keep_attrs else None
+
+        if isinstance(keep_attrs, bool):
+            keep_attrs = "override" if keep_attrs else "drop"
```

Comment:
no, because we need to support both of these:
```suggestion
with xr.set_options(keep_attrs=True):
    ds.mean()
ds.mean(keep_attrs=True)
```
but I can move it to `merge_attrs`